### PR TITLE
Add support for custom value substitution

### PR DESCRIPTION
**Pull request by src-code**
Implements #377, with support for recursive substitutions (maximum depth of 10 substitutions to prevent infinite loops, at which point it throws.)  

cc @snyamathi @redonkulus @carriemorrison @renatoi @thierryk

### CODE REVIEW

File: src/lib/utils.js [hunk 1]
```diff
@@ -52,4 +53,21 @@ utils.repeatString = function (pattern/*:string*/, count/*:integer*/) {
...
 
+// replaces custom value tokens found in a string
+utils.replaceCustomValueTokens = function (config/*:Config*/, value/*:string*/, depth/*:integer*/)/*:string*/ {
+    depth = depth || 1;
+    if (typeof value !== 'string' || value.indexOf('#{') === -1 || !config || !config.custom) {
+        return value;
+    }
+    // Don't allow for infinite loops!
+    if (depth > 10) {
+        throw new Error('Infinite loop detected while substituting custom value tokens. Make sure your custom values don\'t contain tokens that reference one another. Aborting.');
```
Comment (redonkulus) [inline src/lib/utils.js:64]: Would it be helpful to output a breadcrumb of all the variables that were traversed to get to this depth? Like $main-color -> $foo-color -> $bar-color ...

File: tests/atomizer.js [hunk 2]
```diff
@@ -764,6 +764,56 @@ describe('Atomizer()', function () {
...
+            };
+            var expected = [
+                '.Bg\\(my-gradient\\) {',
+                '  background: linear-gradient(to bottom, #000000, #400090);',
+                '}\n'
+            ].join('\n');
+            var result = atomizer.getCss(config);
+            expect(result).to.equal(expected);
+        });
+        it('avoids infinite loops in custom with nested variable substitution', function () {
```
Comment (redonkulus) [inline tests/atomizer.js:804]: Should we add a test for when we go back 10?

**Review by redonkulus — COMMENTED** [review]
This looks really good. Had a couple of comments and questions.

Can we add docs for this somewhere or do we need a follow up PR for the docs site?

File: tests/atomizer.js (hunk 2 above)
Comment (src-code) [reply tests/atomizer.js:804]: Current test uses a looping reference to reach 10

**Comment by src-code** [comment]
New output:

```bash
$ ./bin/atomizer -c config.json test.html
Parsing file test.html for Atomic CSS classes
/Users/yasteve/git/atomizer/src/lib/utils.js:82
        throw new Error('Infinite loop detected while substituting custom value tokens. Make sure your custom values don\'t contain tokens that reference one another. Aborting. Custom value trace: ' + nameStack.join(' > '));
        ^

Error: Infinite loop detected while substituting custom value tokens. Make sure your custom values don't contain tokens that reference one another. Aborting. Custom value trace: my-gradient > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black > $main-color > black
    at Object.utils.getCustomValue (/Users/yasteve/git/atomizer/src/lib/utils.js:82:15)
    at value.replace (/Users/yasteve/git/atomizer/src/lib/utils.js:86:21)
    at String.replace (<anonymous>)
    at Object.utils.getCustomValue (/Users/yasteve/git/atomizer/src/lib/utils.js:85:18)
    at value.replace (/Users/yasteve/git/atomizer/src/lib/utils.js:86:21)
    at String.replace (<anonymous>)
    at Object.utils.getCustomValue (/Users/yasteve/git/atomizer/src/lib/utils.js:85:18)
    at value.replace (/Users/yasteve/git/atomizer/src/lib/utils.js:86:21)
    at String.replace (<anonymous>)
    at Object.utils.getCustomValue (/Users/yasteve/git/atomizer/src/lib/utils.js:85:18)
```

File: tests/atomizer.js (hunk 2 above)
Comment (src-code) [reply tests/atomizer.js:804]: Also, I've increased the limit to 20, just in case.

File: src/lib/utils.js (hunk 1 above)
Comment (src-code) [reply src/lib/utils.js]: Done!

**Review by redonkulus — APPROVED** [review]
woot woot 🎉

**Pull request merged**